## Question :- Give 10 hypothesis statements for "HR Ananlytics Case Study".

In [42]:
%%HTML
<style type="text/css">
table.dataframe td, table.dataframe th {
    border: 1px  black solid !important;
  color: black !important;
}
</style>

In [43]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
data = pd.read_csv("general_data.csv")
data.head(5)

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [44]:
data.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [45]:
data.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   
4409  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   
4409      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   
4409               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   
4409             False               True                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4405           False                    False                 False  
4406           False    

In [20]:
data.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4405    False
4406    False
4407    False
4408    False
4409    False
Length: 4410, dtype: bool

In [21]:
data.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [46]:
data[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion']].describe()

Age  DistanceFromHome    Education  MonthlyIncome  \
count  4410.000000       4410.000000  4410.000000    4410.000000   
mean     36.923810          9.192517     2.912925   65029.312925   
std       9.133301          8.105026     1.023933   47068.888559   
min      18.000000          1.000000     1.000000   10090.000000   
25%      30.000000          2.000000     2.000000   29110.000000   
50%      36.000000          7.000000     3.000000   49190.000000   
75%      43.000000         14.000000     4.000000   83800.000000   
max      60.000000         29.000000     5.000000  199990.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count         4391.000000        4410.000000        4401.000000   
mean             2.694830          15.209524          11.279936   
std              2.498887           3.659108           7.782222   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           6.000000   
50%              2.000000          14.000000          10.000000   
75%              4.000000          18.000000          15.000000   
max              9.000000          25.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  
count            4410.000000     4410.000000              4410.000000  
mean                2.799320        7.008163                 2.187755  
std                 1.288978        6.125135                 3.221699  
min                 0.000000        0.000000                 0.000000  
25%                 2.000000        3.000000                 0.000000  
50%                 3.000000        5.000000                 1.000000  
75%                 3.000000        9.000000                 3.000000  
max                 6.000000       40.000000                15.000000

In [48]:
data[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].sample(30).describe()

Age  DistanceFromHome  Education  MonthlyIncome  \
count  30.000000         30.000000  30.000000      30.000000   
mean   37.233333          8.600000   2.766667   60699.000000   
std     8.459573          9.042429   1.040004   40056.570717   
min    25.000000          1.000000   1.000000   21320.000000   
25%    31.000000          2.000000   3.000000   27895.000000   
50%    36.500000          6.000000   3.000000   50910.000000   
75%    40.750000          9.000000   3.000000   74732.500000   
max    55.000000         28.000000   5.000000  166590.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count           30.000000          30.000000          30.000000   
mean             3.333333          15.433333          13.366667   
std              2.844636           3.370085           8.798446   
min              0.000000          11.000000           1.000000   
25%              1.000000          13.000000           7.250000   
50%              2.500000          14.000000          10.500000   
75%              5.500000          18.000000          16.500000   
max              9.000000          22.000000          36.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              30.000000       30.000000                30.000000   
mean                2.866667        7.633333                 2.000000   
std                 1.105888        4.708637                 2.626129   
min                 1.000000        1.000000                 0.000000   
25%                 2.000000        5.250000                 0.250000   
50%                 3.000000        6.500000                 1.000000   
75%                 3.000000       10.000000                 2.750000   
max                 6.000000       22.000000                11.000000   

       YearsWithCurrManager  
count             30.000000  
mean               5.433333  
std                3.568863  
min                0.000000  
25%                3.000000  
50%                5.000000  
75%                7.750000  
max               15.000000


## Hypothesis 1
Null Hypothesis:<br>
The mean age of an employee who has attrited is found to be 33 with a standard deviation of 9.67.<br>

Alternative Hypothesis:<br>
The mean age of an employee who has attrited is not equal to 33.<br>

To determine if this is true, a sample of 30 attrited employees was taken and their mean age was found to be 37.233 with a standard deviation of 8.458.

In [33]:
df_1 = data.loc[data['Attrition'] == 'Yes']
df_1_desc=df_1[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()
df_1_desc

Age  DistanceFromHome   Education  MonthlyIncome  \
count  711.000000        711.000000  711.000000     711.000000   
mean    33.607595          9.012658    2.877637   61682.616034   
std      9.675693          7.772368    1.014233   44792.067695   
min     18.000000          1.000000    1.000000   10090.000000   
25%     28.000000          2.000000    2.000000   28440.000000   
50%     32.000000          7.000000    3.000000   49080.000000   
75%     39.000000         15.000000    4.000000   71040.000000   
max     58.000000         29.000000    5.000000  198590.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count          707.000000         711.000000         709.000000   
mean             2.936351          15.481013           8.255289   
std              2.678774           3.775289           7.164018   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           3.000000   
50%              1.000000          14.000000           7.000000   
75%              5.000000          18.000000          10.000000   
max              9.000000          25.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             711.000000      711.000000               711.000000   
mean                2.654008        5.130802                 1.945148   
std                 1.154834        5.941598                 3.148633   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        1.000000                 0.000000   
50%                 3.000000        3.000000                 1.000000   
75%                 3.000000        7.000000                 2.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count            711.000000  
mean               2.852321  
std                3.138918  
min                0.000000  
25%                0.000000  
50%                2.000000  
75%                5.000000  
max               14.000000

In [37]:
df_1[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].sample(30).describe()

Age  DistanceFromHome  Education  MonthlyIncome  \
count  30.000000         30.000000  30.000000      30.000000   
mean   35.633333          9.433333   2.700000   63354.000000   
std    10.597603          7.885750   1.118805   43842.426751   
min    19.000000          1.000000   1.000000   20700.000000   
25%    28.000000          2.000000   2.000000   34365.000000   
50%    33.000000          8.000000   3.000000   49520.000000   
75%    40.750000         15.000000   3.000000   68145.000000   
max    58.000000         25.000000   5.000000  189470.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count           30.000000          30.000000          30.000000   
mean             3.033333          14.166667           8.766667   
std              2.684352           3.249226           7.855081   
min              0.000000          11.000000           1.000000   
25%              1.000000          11.000000           5.000000   
50%              2.000000          13.500000           7.000000   
75%              4.750000          15.750000          10.750000   
max              9.000000          22.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              30.000000       30.000000                30.000000   
mean                2.766667        5.866667                 2.133333   
std                 0.817200        7.833321                 3.298206   
min                 1.000000        0.000000                 0.000000   
25%                 2.000000        1.250000                 0.000000   
50%                 3.000000        3.000000                 1.000000   
75%                 3.000000        7.000000                 2.750000   
max                 5.000000       40.000000                15.000000   

       YearsWithCurrManager  
count              30.00000  
mean                2.50000  
std                 2.77613  
min                 0.00000  
25%                 0.00000  
50%                 2.00000  
75%                 5.25000  
max                 8.00000

## Hypothesis 2
Null Hypothesis:<br>
The mean monthly income of attrited employees was found to be Rs.61,000 with a standard deviation of 44792.

Alternative Hypothesis:<br>
The mean monthly income of attrited employees is not Rs.61,000.<br>

To determine if this is true, a sample of 30 attrited employees was taken and their mean monthly income was found to be Rs.63,354.

## Hypothesis 3
Null Hypothesis:<br>
The mean distance from home of attrited employees was found to be greater than 9.<br>
<br>Alternative Hypothesis:<br>
The mean distance from home of attrited employees is not greater than 9.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean distance from home was found to be 9.433. 

## Hypothesis 4
Null Hypothesis:<br>
Mean of total working years of attrited employees was found to be less than or equal to 10 years with a standard deviation of 7.16.<br>
<br>Alternative Hypothesis:<br>
Mean of total working years of attrited employees is greater than 10 years.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean working years was found to be 8.766 years.

## Hypothesis 5
Null Hypothesis:<br>
Mean of year since last promotion of attrited employees was found to be less than 2 years.<br>
<br>Alternative Hypothesis:<br>
Mean of year since last promotion of attrited employees is not less than 2 years.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean of year since last promotion was found to be 2.133.

## Hypothesis 6
Null Hypothesis:<br>
The mean years at company of attrited employees was found to be greater than or equal to 4 years with a standard deviation of 5.94</br>

Alternative Hypothesis:<br>
The mean years at company of attrited employees is less than 4 years.<br>

To determine if this is true,a sample of 30 attrited employees was taken at random and their mean years at the comapny was found to be 5.866 years

## Hypothesis 7
Null Hypothesis:<br>
The mean pecentage in salary hike of the attrited employees is found to be greater than or equal to 15%<br>

Alternative Hypothesis:<br>
The mean pecentage in salary hike of the attrited employees is found to be less than 15%.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean percentage in salary hike was found to be 14.167%

In [38]:
df_2 = data.loc[data['Attrition'] == 'No']
df_2[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()

Age  DistanceFromHome    Education  MonthlyIncome  \
count  3699.000000       3699.000000  3699.000000    3699.000000   
mean     37.561233          9.227088     2.919708   65672.595296   
std       8.885956          8.167978     1.025784   47472.814021   
min      18.000000          1.000000     1.000000   10510.000000   
25%      31.000000          2.000000     2.000000   29360.000000   
50%      36.000000          7.000000     3.000000   49300.000000   
75%      43.000000         14.000000     4.000000   86060.000000   
max      60.000000         29.000000     5.000000  199990.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count         3684.000000        3699.000000        3692.000000   
mean             2.648480          15.157340          11.860780   
std              2.460537           3.634551           7.762536   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           6.000000   
50%              2.000000          14.000000          10.000000   
75%              4.000000          18.000000          16.000000   
max              9.000000          25.000000          38.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count            3699.000000     3699.000000              3699.000000   
mean                2.827251        7.369019                 2.234388   
std                 1.311493        6.094649                 3.233887   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        3.000000                 0.000000   
50%                 3.000000        6.000000                 1.000000   
75%                 3.000000       10.000000                 3.000000   
max                 6.000000       37.000000                15.000000   

       YearsWithCurrManager  
count           3699.000000  
mean               4.367397  
std                3.593144  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               17.000000

In [41]:
df_2[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].sample(30).describe() 

Age  DistanceFromHome  Education  MonthlyIncome  \
count  30.000000         30.000000  30.000000      30.000000   
mean   37.166667          6.433333   2.900000   65211.666667   
std     8.321486          7.375090   0.922889   45440.126433   
min    20.000000          1.000000   1.000000   22310.000000   
25%    31.000000          1.000000   2.250000   33315.000000   
50%    35.500000          2.000000   3.000000   48635.000000   
75%    42.750000          9.750000   3.750000   84617.500000   
max    55.000000         26.000000   4.000000  195170.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count           29.000000          30.000000          30.000000   
mean             2.000000          14.866667          12.133333   
std              2.017778           3.431430           7.573699   
min              0.000000          11.000000           1.000000   
25%              1.000000          12.000000           7.000000   
50%              1.000000          13.000000          10.000000   
75%              2.000000          17.000000          16.750000   
max              9.000000          24.000000          29.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count              30.000000       30.000000                30.000000   
mean                2.633333        8.733333                 2.533333   
std                 1.066200        6.285908                 3.148435   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        4.250000                 0.250000   
50%                 3.000000        7.500000                 2.000000   
75%                 3.000000       11.000000                 3.750000   
max                 5.000000       27.000000                15.000000   

       YearsWithCurrManager  
count             30.000000  
mean               6.033333  
std                4.444938  
min                0.000000  
25%                2.250000  
50%                7.000000  
75%                8.000000  
max               17.000000

## Hypothesis 8
Null Hypothesis:<br>
The mean monthly income of non-attrited employees was found to be Rs.65,000 with a standard deviation of 44792.

Alternative Hypothesis:<br>
The mean monthly income of non-attrited employees is not Rs.65,000.<br>

To determine if this is true, a sample of 30 attrited employees was taken and their mean monthly income was found to be Rs.665,211.667.

## Hypothesis 9
Null Hypothesis:<br>
Mean of total working years of non-attrited employees was found to be less than or equal to 10 years with a standard deviation of 7.16.<br>
<br>Alternative Hypothesis:<br>
Mean of total working years of non-attrited employees is greater than 10 years.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean working years was found to be 12.133 years.

## Hypothesis 10
Null Hypothesis:<br>
The mean pecentage in salary hike of the non-attrited employees is found to be greater than or equal to 15%<br>

Alternative Hypothesis:<br>
The mean pecentage in salary hike of the attrited employees is found to be less than 15%.<br>

In order to test this claim, a sample of 30 employees who have attrited was taken at random and their mean percentage in salary hike was found to be 14.867%